src/main/java/com/example/springbootbookshop/controller/CategoryController.java

```diff
@@ -0,0 +1,83 @@
+package com.example.springbootbookshop.controller;
+
+import com.example.springbootbookshop.dto.book.BookDtoWithoutCategoryIds;
+import com.example.springbootbookshop.dto.category.CategoryDto;
+import com.example.springbootbookshop.dto.category.CategoryRequestDto;
+import com.example.springbootbookshop.entity.Category;
+import com.example.springbootbookshop.service.BookService;
+import com.example.springbootbookshop.service.CategoryService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.Positive;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@Tag(name = "Categories management",description = "Endpoints for managing categories")
+@RestController
+@RequestMapping(value = "/category")
+public class CategoryController {
+    private final CategoryService categoryService;
+    private final BookService bookService;
+
+    @PostMapping
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @ResponseStatus(HttpStatus.OK)
+    @Operation(summary = "Create new category (for admin only)")
+    public Category createCategory(@RequestBody @Valid CategoryDto categoryDto) {
+        return categoryService.save(categoryDto);
+    }
+
+    @GetMapping
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @ResponseStatus(HttpStatus.OK)
+    @Operation(summary = "Get all categories from DB")
+    public List<CategoryDto> getAll() {
+        return categoryService.findAll();
+    }
+
+    @GetMapping(value = "/{id}")
+    @ResponseStatus(HttpStatus.OK)
+    @Operation(summary = "Get category by id from DB")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    public CategoryDto getCategoryById(@PathVariable @Positive Long id) {
```

Comment:
Your admin can't get all categories in this case
```suggestion
    @PreAuthorize("hasAnyRole('ROLE_USER', 'ROLE_ADMIN')")
    public CategoryDto getCategoryById(@PathVariable @Positive Long id) {
```

Reply:
@OlehReex no, this is wrong

admin should have all the priviliges a regular user has

in other words admin user would have authorities like `["USER", "ADMIN"]`